api/types/network/network.go

```diff
@@ -28,6 +28,14 @@ type EndpointIPAMConfig struct {
 	LinkLocalIPs []string `json:",omitempty"`
 }
 
+// Copy makes a copy of the endpoint ipam config
+func (cfg *EndpointIPAMConfig) Copy() *EndpointIPAMConfig {
```

Comment:
I think this is very difficult to do correctly, particularly when non-exported types/fields come into play.